### PR TITLE
Implement resource bundle depends on `all` to be rendered at last

### PR DESCRIPTION
**Pull request by petschki**
This is a successor of #4054 with the simplified `webresource.ResourceGroup` approach.
Tests are copied 1:1 from the previous PR

### CODE REVIEW

**Comment by petschki** [comment]
@jenkins-plone-org please run jobs

**Review by ale-rt — APPROVED** [review]
LGTM

File: Products/CMFPlone/resources/browser/resource.py [hunk 1]
```diff
@@ -146,6 +152,11 @@ def check_dependencies(bundle_name, depends, bundles):
                 if depends == "__broken__":
                     continue
                 external = self.is_external_url(record.jscompilation)
+                r_group = registry_group_js
+                if "all" in depends:
+                    # move to a separate group which is rendered at last
```
Comment (thet) [inline Products/CMFPlone/resources/browser/resource.py:157]:
```suggestion
                    # move to a separate group which is rendered after all others
```

File: Products/CMFPlone/resources/browser/resource.py [hunk 2]
```diff
@@ -168,6 +179,11 @@ def check_dependencies(bundle_name, depends, bundles):
                 if depends == "__broken__":
                     continue
                 external = self.is_external_url(record.csscompilation)
+                r_group = registry_group_css
+                if "all" in depends:
+                    # move to a separate group which is rendered at last
```
Comment (thet) [inline Products/CMFPlone/resources/browser/resource.py:184]:
```suggestion
                    # move to a separate group which is rendered after all others
```

File: Products/CMFPlone/resources/browser/resource.py [hunk 3]
```diff
@@ -237,7 +253,11 @@ def check_dependencies(bundle_name, depends, bundles):
                 **{"data-bundle": "diazo"},
             )
 
-        # add Custom CSS
+        # add "deferred" groups at this point
+        root_group_js.add(registry_group_js_deferred)
+        root_group_css.add(registry_group_css_deferred)
+
+        # add Custom CSS always after everything
```
Comment (thet) [inline Products/CMFPlone/resources/browser/resource.py:260]:
```suggestion
        # add Custom CSS always after everything else
```

File: news/4054.feature [hunk 4]
```diff
@@ -0,0 +1,18 @@
+Allow bundles to be rendered after all other.
```
Comment (thet) [inline news/4054.feature:1]:
```suggestion
Allow bundles to be rendered after all others.
```

File: news/4054.feature [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+JS and CSS resources can now be rendered after all other resources in their
+resource group including the theme (e.g. the Barceloneta theme CSS).
+
+There is a exception for custom CSS which can be defined in the theming
```
Comment (thet) [inline news/4054.feature:6]:
```suggestion
There is an exception for custom CSS which can be defined in the theming
```

File: news/4054.feature [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+controlpanel. This one is always rendered as last style resource.
+
+To render a resource after all other give it the "depends" value of "all". This
```
Comment (thet) [inline news/4054.feature:9]:
```suggestion
To render resources after all others, give them the "depends" value of "all".
```

File: news/4054.feature [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+indicates that the resource depends on all other being rendered before. The
```
Comment (thet) [inline news/4054.feature:10]:
```suggestion
For each of these resources, "all" indicates that the resource depends on all other resources, making it render after its dependencies.
```

File: news/4054.feature [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+resource is then rendered as last resource of it's resource group.
```
Comment (thet) [inline news/4054.feature:11]:
```suggestion
If you set multiple resources with "all", then they will render alphabetically after all other.
```

File: news/4054.feature [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+This allows to override a theme with custom CSS from a bundle instead of having
```
Comment (thet) [inline news/4054.feature:13]:
```suggestion
This lets you override a theme with custom CSS from a bundle instead of having
```

**Review by thet — APPROVED** [review]
Thanks for the improvement.

My suggestions are basically @stevepiercy suggestions from:
https://github.com/plone/Products.CMFPlone/pull/4074

**Pull request merged**